framework/src/st_things/things_stack/things_stack.c

```diff
@@ -298,7 +300,16 @@ int things_initialize_stack(const char *json_path, bool *easysetup_completed)
 	}
 	is_things_module_inited = 1;
 
-	*easysetup_completed = dm_is_there_things_cloud();
+	*easysetup_completed = (int) esm_read_easysetup_state();
+
+	if (esm_read_easysetup_state() != 1) {
```

Comment:
You don't need to call esm_read_easysetup_state() twice.
I want you to use a local variable to store the result of esm_read_easysetup_state() at line 303.

Reply:
You are right.
I will change it to use the above variables.

Reply:
The type of *easysetup_completed is bool. You need to use the same type.